### PR TITLE
Use String::replaceAll only when fist parameter is regular expression

### LINKED ISSUE #1660: `String.replace` should be preferred to `String.replaceAll` if the first parameter doesn’t contains a regular expression or pattern. `RSPEC-5361`
https://rules.sonarsource.com/java/RSPEC-5361

### PR DESCRIPTION
**Pull request by anelfdzml**
## Description
This PR tries to fix the [issue](https://github.com/openrewrite/rewrite/issues/1660). 

### Recipe description
When `String::replaceAll` is used, the first argument should be a real regular expression. If it’s not the case, `String::replace` does exactly the same thing as `String::replaceAll` without the performance drawback of the regex.

fixes #1660

### CODE REVIEW

File: rewrite-java/src/main/java/org/openrewrite/java/cleanup/StringAvoidReplaceAll.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+ * Recipe to avoid using {@link String#replaceAll(String, String)} when the fist argument is not a regular expression.
+ * <p>
+ * The underlying implementation of {@link String#replaceAll(String, String)} calls the {@link Pattern#compile(String)}
+ * method each time it is called even if the first argument is not a regular expression. This has a significant
+ * performance cost and therefore should be used with care.
+ *
+ * @see <a href="https://rules.sonarsource.com/java/RSPEC-5361"></a>
+ * @see <a href="https://docs.oracle.com/javase/tutorial/essential/regex/index.html"></a>
+ */
+public class StringAvoidReplaceAll extends Recipe {
```
Comment (traceyyoshima) [inline rewrite-java/src/main/java/org/openrewrite/java/cleanup/StringAvoidReplaceAll.java:44]: Hi @anelfdzml , thank you for the PR!

I have a few suggestions to fit our general naming practices. It's preferred to name recipes based on the action it's performing like `UseStringReplace`.

Similarly in the `displayName`, "Use `String::replace` when.

File: rewrite-java/src/main/java/org/openrewrite/java/cleanup/StringAvoidReplaceAll.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+    @Override
+    public Set<String> getTags() {
+        return Collections.singleton("RSPEC-5361");
+    }
+
+    @Override
+    public Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(10);
```
Comment (traceyyoshima) [inline rewrite-java/src/main/java/org/openrewrite/java/cleanup/StringAvoidReplaceAll.java:64]: I looked up the RSPEC and the time estimate is 2min per occurence.

File: rewrite-test/src/main/kotlin/org/openrewrite/java/cleanup/StringAvoidReplaceAllTest.kt [hunk 3]
```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2021 the original author or authors.
```
Comment (traceyyoshima) [inline rewrite-test/src/main/kotlin/org/openrewrite/java/cleanup/StringAvoidReplaceAllTest.kt:2]: Minor update from `2021` to `2022`

File: rewrite-java/src/main/java/org/openrewrite/java/cleanup/StringAvoidReplaceAll.java [hunk 4]
```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2021 the original author or authors.
```
Comment (traceyyoshima) [inline rewrite-java/src/main/java/org/openrewrite/java/cleanup/StringAvoidReplaceAll.java:2]: Minor update from `2021` to `2022`

**Review by traceyyoshima — CHANGES_REQUESTED** [review]

File: rewrite-java/src/main/java/org/openrewrite/java/cleanup/StringAvoidReplaceAll.java (hunk 1 above)
Comment (murdos) [inline rewrite-java/src/main/java/org/openrewrite/java/cleanup/StringAvoidReplaceAll.java:44]: Hi @anelfdzml,

Could you also include this recipe in the compound `CommonStaticAnalysis` recipe: https://github.com/openrewrite/rewrite/blob/main/rewrite-java/src/main/resources/META-INF/rewrite/static-analysis-cleanup.yml

Comment (anelfdzml) [reply rewrite-java/src/main/java/org/openrewrite/java/cleanup/StringAvoidReplaceAll.java]: I changed the name as you suggested!

File: rewrite-java/src/main/java/org/openrewrite/java/cleanup/StringAvoidReplaceAll.java (hunk 2 above)
Comment (anelfdzml) [reply rewrite-java/src/main/java/org/openrewrite/java/cleanup/StringAvoidReplaceAll.java]: Yes, you're right! I changed the value

File: rewrite-test/src/main/kotlin/org/openrewrite/java/cleanup/StringAvoidReplaceAllTest.kt (hunk 3 above)
Comment (anelfdzml) [reply rewrite-test/src/main/kotlin/org/openrewrite/java/cleanup/StringAvoidReplaceAllTest.kt]: Nice catch!

File: rewrite-java/src/main/java/org/openrewrite/java/cleanup/StringAvoidReplaceAll.java (hunk 1 above)
Comment (anelfdzml) [reply rewrite-java/src/main/java/org/openrewrite/java/cleanup/StringAvoidReplaceAll.java]: Of course, I included it!

**Review by traceyyoshima — APPROVED** [review]
LGTM, thank you for the PR!

**Pull request merged**